### DISCUSSION TITLE
Graph RAG API doesn't have any way to insert openAI keys and there is no model selection feature.

### DISCUSSION THREAD

**Discussion by tanusri9494** [Azure-Samples/graphrag-accelerator#124] [General]
Hey i am trying to access graph rag API in node js. 
I gave the **Subscription keys** by **creating a API management resource group** and while i am trying to index it doesn't actually work. 

I thought may be it requires Openai api keys. so i check, and i find there is no way mentioned in the api docs that allows me to Insert Openai keys with an option to select model on my own :)))) 

[https://github.com/Azure-Samples/graphrag-accelerator/blob/main/notebooks/1-Quickstart.ipynb](Quickstart)
https://github.com/Azure-Samples/graphrag-accelerator/blob/c211508a7d569ea71f831a171cc5cbd832c69264/notebooks/1-Quickstart.ipynb

this is really disappointing plz help :) @chanezon @erikstmartin

**Comment by timothymeyers** [comment]
Hi @tanusri9494 - it is the APIM subscription key that you should use. You are not interacting directly with OpenAI / the LLM so you wouldn't need the OpenAI API key; that all is handled by the `graphrag` services running in AKS that you go through APIM to reach.

This example code is designed to run with the Azure OpenAI Service running within the same Azure subscription it is deployed into. See [here](https://github.com/Azure-Samples/graphrag-accelerator/blob/main/docs/DEPLOYMENT-GUIDE.md#1-deploy-azure-openai-service).

> _Note that the AOAI instance must be in the same subscription that you plan to deploy this solution accelerator in._

Inside the Azure OpenAI Service you can choose from a number of different models and configure your deployment accordingly. We've given our recommended models (ada02 for embeddings and gtp-4 turbo, at least, for LLM) and settings in the Deployment guide.

While I am sure you can dive in and update the [backend](https://github.com/Azure-Samples/graphrag-accelerator/tree/main/backend) service to use a non-Azure OpenAI Service LLM, it will take some work.

**Reply by tanusri9494** [reply]
Also i am having AIPM resource in Developer tier, so is it the root cause of the graph rag apis not working for me? 

do i have to switch a paid production tier to make i work??

**Comment by tanusri9494** [comment]
hey hi really thanks, so you mean i dont need to worry about the openai apis its all done already for me. 

So this is what i did, i created a Api management resource named graphrag and in that i selected developer role fee tire.
And i then try to connect with the endpoint with a subscription Primary key. 

@timothymeyers  but it returns me a empty response when i try to query the rag, it just doesn't return anything 

![image](https://github.com/user-attachments/assets/a644ca2a-9a0e-4ca9-a40c-d4d19823397b)

![image](https://github.com/user-attachments/assets/4895d1c9-07ad-4bb1-a92e-95372007a26e)

**Comment by timothymeyers** [comment]
@tanusri9494 - It looks like you may have used the **Developer Portal URL** instead of the **Gateway URL** when configuring your GraphRAG deployment.

That is the standard html response for the Dev portal that has not been "published" yet

You can update your `deploy.parameters.json` with the Gateway URL, or you can update the config map inside AKS directly (more advanced).

I also confirmed that the Accelerator works with the Developer sku for APIM this morning. It is actually going to be the default APIM SKU after #123 is merged.

I also wanted to confirm with you that you followed the Deployment Guide and deployed everything using `deploy.sh`? After reading a few of your comments above, I was not exactly sure how you deployed it.

**Reply by timothymeyers** [reply]
#123 was just merged, FYI. @tanusri9494 ok if we close out this Discussion?
